### ISSUE TITLE
llvm build fails on Mac after llvm commit update

### ISSUE THREAD

**Issue by sorenlassen** [onnx/onnx-mlir#1619]
UPDATE: the originally reported issue is no longer reproducible but there's a new build issue reported in the comments below

after updating the llvm commit in PR #1599 and follow the instructions in [BuildOnLinuxOSX.md](https://github.com/onnx/onnx-mlir/blob/main/docs/BuildOnLinuxOSX.md):
```
mkdir llvm-project/build
cd llvm-project/build
cmake -G Ninja ../llvm \
   -DLLVM_ENABLE_PROJECTS=mlir \
   -DLLVM_TARGETS_TO_BUILD="host" \
   -DCMAKE_BUILD_TYPE=Release \
   -DLLVM_ENABLE_ASSERTIONS=ON \
   -DLLVM_ENABLE_RTTI=ON

cmake --build . -- ${MAKEFLAGS}
```
I experience the following compile error on Mac:
```
/Users/sorenlassen/git/llvm-project/mlir/include/mlir/IR/OpDefinition.h:1497:5: error: no template named 'disjunction' in namespace 'std'; did you mean 'llvm::disjunction'?
    std::disjunction<detect_has_fold_trait<T>,
    ^~~~~~~~~~~~~~~~
    llvm::disjunction
/Users/sorenlassen/git/llvm-project/llvm/include/llvm/ADT/STLForwardCompat.h:41:8: note: 'llvm::disjunction' declared here
struct disjunction // NOLINT(readability-identifier-naming)
```
plus a bunch of "... is a C++17 extension" warnings

**Comment by chentong319** [comment]
I got different compilation error on my Mac:
```
/Users/tongchen/Projects/llvm-project/mlir/lib/Bytecode/Reader/BytecodeReader.cpp:169:60: error: non-const lvalue reference to type 'uint64_t' (aka 'unsigned long long') cannot bind to a value of unrelated type 'size_t' (aka 'unsigned long')
    if (failed(parseByte(sectionID)) || failed(parseVarInt(length)))
                                                           ^~~~~~
/Users/tongchen/Projects/llvm-project/mlir/lib/Bytecode/Reader/BytecodeReader.cpp:110:39: note: passing argument to parameter 'result' here
  LogicalResult parseVarInt(uint64_t &result) {
```

**Comment by sorenlassen** [comment]
I'm not seeing the C++17 problems anymore (maybe they went away when with the further llvm update in PR #1628?) but now I'm seeing the same problem that @chentong319 reported

it seems like it was solved in LLVM in commit [59548fe873d8d98e359fb21fbb2a0852fed17ff5](https://github.com/llvm/llvm-project/commit/59548fe873d8d98e359fb21fbb2a0852fed17ff5)

**Comment by tungld** [comment]
Thanks @sorenlassen!
I created this PR #1630 to see if [59548fe873d8d98e359fb21fbb2a0852fed17ff5](https://github.com/llvm/llvm-project/commit/59548fe873d8d98e359fb21fbb2a0852fed17ff5) works.

**Comment by chentong319** [comment]
I fixed my error by changing the type of that variable in BytecodeReader.cpp. I think it is an issue in MLIR. Since the function requires uint64_t argument, the variable should be uint64_t instead of size_t.

**Comment by sorenlassen** [comment]
solved by PR #1630
